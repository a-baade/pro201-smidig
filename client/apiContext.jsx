import { fetchJSON } from "./lib/fetchJSON";
import { postJSON } from "./lib/postJSON";
import React from "react";

export const ApiContext = React.createContext({
  async fetchLogin() {
    return await fetchJSON("/api/login");
  },

  async allCharities() {
    return await fetchJSON("/api/charities");
  },

  async registerLogin(provider, login) {
    return await postJSON(`/api/login/${provider}`, login);
  },
  async endSession() {
    const res = await fetch("/api/login", { method: "DELETE" });
    if (!res.ok) {
      throw new Error(`Failed to post ${res.status}: ${res.statusText}`);
    }
  },
});