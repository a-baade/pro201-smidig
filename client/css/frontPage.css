/* Style for frontPage.jsx */

.front-image {
    display: flex;
    justify-content: flex-end;
    align-items: center;
    flex-direction: column;
    height: 90vh;
    background-position: top;
}

.front-image-text {
    text-align: center;
    font-size: 64px;
    text-shadow: 0 0 10px #333232;
    color: white;
    margin-bottom: 10vh;
}

.front-image-button {
    z-index: 99;
    font-size: 14px;
    font-weight: 500;
    text-transform: uppercase;
    letter-spacing: 2px;
    background-color: #8AB7BA;
    border-radius: 10px;
    border: 0;
    min-width: 150px;
    width: 20vw;
    padding: 15px 30px;
    margin-bottom: 25vh;
}
.about-us-title {
    text-align: center;
}

.about-us-div {
    position: relative;
    margin: 80px;
    padding: 12px 25px;
    padding-inline: 0.5em;
    background-color: #95BDC4;
    border-radius: 5px;
    box-sizing: border-box;
    text-align: left;
    line-height: 25px;
    font-size: 18px;
}
.about-us-div h1 {
    margin-bottom: 5vh;
}

.about-us-div p {
    padding: 35px;
}

.about-us-button {
    position: relative;
    left: 45%;
    margin: 3vh 10px 10px 10px;
    padding: 17px 32px;
    border-radius: 10px;
    border: 0;
    background: #d5e5e6;
    text-transform: uppercase;
    letter-spacing: 2px;
    font-size: 12px;
    font-weight: 600;

}

.topic-container {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    gap: 50px;
    text-align: center;
    margin: 80px;
}

.topic h4 {
    padding: 10px;
    font-weight: bold;
    font-size: 27px;
}

.topic-image img {
    width: 300px;
    height: 300px;
    border-radius: 50%;
}

.topic p {
    width: 100%;
    min-width: 320px;
    max-width: 400px;
}

.newsfeed-container {
    background-color: #95BDC4;
    margin: 80px;
    padding: 12px 20px;
    padding-inline: 0.5em;
    box-sizing: border-box;
    border-radius: 5px;
    text-align: center;
    font-size: 20px;
    line-height: 25px;
}

.newsfeed-items-container {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    gap: 10px;
    margin: 80px 40px;
    box-sizing: border-box;
    text-align: center;
    font-size: 14px;
}

.newsfeed-item {
    box-sizing: content-box;
    background-color: white;
    border-radius: 5px;
    padding-bottom: 10px;
    width: 30%;
    min-width: 250px;
    height: auto;
    min-height: 300px;
    display: flex;
    flex-direction: column;
}

.newsfeed-image {
    object-fit: cover;
    width: 100%;
    height: 250px;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
}

.newsfeed-item p {
    margin: 5px;
    text-align: left;
    padding: 15px;
}

.read-more {
    display: flex;
    height: 100%;
    width: 100%;
    justify-content: center;
    align-items: end;
    font-weight: bold;
    text-decoration: underline;
    text-align: center;
    cursor: pointer;
}