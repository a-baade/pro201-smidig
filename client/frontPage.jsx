
import React from "react";


export function FrontPage() {

    return (
      <div>
       <h1>Front page</h1>
      </div>
    );
}
