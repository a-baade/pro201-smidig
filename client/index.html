<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Meliora Impact</title>
    <link rel="icon" href="pics/icons/logo.png">
    <link rel="stylesheet" href="css/navigation.css">
    <link rel="stylesheet" href="css/style.css">
    <link rel="stylesheet" href="css/frontPage.css">
    <link rel="stylesheet" href="css/login.css">
    <link rel="stylesheet" href="css/register.css">
    <link rel="stylesheet" href="css/aboutUs.css">
    <link rel="stylesheet" href="css/signup.css">
    <link rel="stylesheet" href="css/charities.css">
    <link rel="stylesheet" href="css/charityPage.css">
    <link rel="stylesheet" href="css/donate.css">
    <link rel="stylesheet" href="css/profile.css">
    <link rel="stylesheet" href="css/contact.css">
    <link rel="stylesheet" href="css/dashboard.css">
    <link rel="stylesheet" href="css/loadingWheel.css">
    <link rel="stylesheet" href="css/footer.css">
</head>
<body>
<div id="app"></div>
</body>
<script src="index.jsx" type="module"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/react/15.1.0/react.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/react/15.1.0/react-dom.min.js"></script>
</html>