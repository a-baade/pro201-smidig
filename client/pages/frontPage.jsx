import React from "react";
import FrontBg from "../pics/bg-pictures/water-pic.png";
import Nature from "../pics/bg-pictures/img.png";
import Welfare from "../pics/bg-pictures/img_1.png";
import Education from "../pics/bg-pictures/img_2.png";
import NewsImg1 from "../pics/bg-pictures/img_3.png";
import NewsImg2 from "../pics/bg-pictures/img_4.png";
import NewsImg3 from "../pics/bg-pictures/img_5.jpg";
import { Link } from "react-router-dom";

export function FrontPage() {
  return (
    <div className={"page-margin"}>
      <div
        className={"front-image"}
        style={{ backgroundImage: `url(${FrontBg})` }}
      >
        <h1 className={"front-image-text custom-font-knowledge"}>
          Meliora Impact
        </h1>
        <Link to={"/signup"}>
          <button className={"front-image-button custom-shadow-button"}>
            Join us
          </button>
        </Link>
      </div>

      <div className={"about-us-div"}>
        <div className={"about-us-title"}>
          <h1>About us</h1>
        </div>
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur
          viverra, eros at luctus molestie, dolor risus convallis orci, ut
          tempus tellus nunc quis nulla. Maecenas libero lectus, pharetra ut
          magna in, rutrum luctus lorem. Nulla blandit augue nulla, sit amet
          porta dolor interdum eu. Phasellus suscipit porttitor urna, non
          vestibulum dui volutpat quis. Donec viverra lectus diam, sed
          sollicitudin quam pretium at. Curabitur nisl ligula, porta nec finibus
          eu, venenatis vitae quam. Curabitur facilisis magna condimentum
          maximus feugiat. Aliquam erat volutpat. Mauris posuere vel mi non
          commodo. Nunc dapibus libero ipsum, placerat posuere augue vulputate
          vitae. Aenean volutpat sollicitudin accumsan. Vestibulum at maximus
          purus, in hendrerit eros. Phasellus id tellus a eros volutpat rutrum
          et in ligula. Donec quis odio augue. Mauris pellentesque feugiat lacus
          eget ornare.
        </p>
        <Link to={"/about"}>
          <button className={"about-us-button custom-shadow-button"}>
            Read more
          </button>
        </Link>
      </div>
      <div className={"topic-container"}>
        <div className={"topic"}>
          <h4>Protecting nature</h4>
          <div className={"topic-image"}>
            <img src={Nature} alt={"Picture of nature"} />
          </div>
          <p>
            Nature is by design sustainable, it has supported itself for decades
            but human intervention across the past centuries has been weakening
            the very infrastructure it stands on. We always aim to protect and
            invest in what is already here so we as well it stands a chance for
            future growth.
          </p>
        </div>
        <div className={"topic"}>
          <h4>Welfare</h4>
          <div className={"topic-image"}>
            <img src={Welfare} alt={"Picture of poor children"} />
          </div>
          <p>
            The clean water crisis persist as a leading issue in several large
            parts of the world. It is a human right and necessity yet is
            consistently overlooked and lacking in support. Our aim is to uplift
            and highlight these core violations of human rights to foster more
            support for the communities at large.
          </p>
        </div>
        <div className={"topic"}>
          <h4>Education</h4>
          <div className={"topic-image"}>
            <img src={Education} alt={"Picture of child in school"} />
          </div>
          <p>
            Studies prove that education plays a large role in how well people
            in communities of all ranges fare supporting themselves and breaking
            cycles generational poverty and trauma. We strongly believe it is a
            right to have access to quality education and learning which is why
            it is part of our focus.
          </p>
        </div>
      </div>
      <div className={"newsfeed-container custom-shadow"}>
        <h1>NEWSFEED</h1>
        <div className={"newsfeed-items-container"}>
          <div className={"newsfeed-item custom-shadow"}>
            <img
              src={NewsImg1}
              className={"newsfeed-image"}
              alt={"Afghan children in school"}
            />
            <p>
              Afghan refugee children enrolling in norwegian nurseries 2 weeks
              after coming to norway. Prime minister hopes that these efforts
              will help stabilize the everyday life of the children and that
              they will integrate quicker.
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
          <div className={"newsfeed-item custom-shadow"}>
            <img
              src={NewsImg2}
              className={"newsfeed-image"}
              alt={"Deers in nature"}
            />
            <p>
              Hunting season has ended, yet people are still hunting animals
              illegally. Protecting nature is our responsibility, and make sure
              that the next generation can enjoy wilderness like us.
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
          <div className={"newsfeed-item custom-shadow"}>
            <img
              src={NewsImg3}
              className={"newsfeed-image"}
              alt={"Akeem in karate training"}
            />
            <p>
              Thanks to the support of charities, children like Akeem are able
              to partake in after school activities. Children like Akeem would
              not be able to afford ...
            </p>
            <p className={"read-more"}>Read more</p>
          </div>
        </div>
      </div>
    </div>
  );
}
