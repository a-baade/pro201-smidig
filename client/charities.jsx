import React, { useContext } from "react";
import { ApiContext } from "./apiContext";
import { useLoading } from "./useLoading";

function CharityCard({charity: {name, description}}){
  return (
    <div class="flex-container">
      <div className={"charity-card"}>
        <h3>{name}</h3>
        <p>{description}</p>
        <div className={"card-buttons"}>
          <button className="card-btn-left card-btn">Read more</button>
          <button className="card-btn-right card-btn">Donate</button>
        </div>
      </div>

    </div>
  );
}

export function Charities() {
  const {allCharities} = useContext(ApiContext);
  const {loading, error, data} = useLoading(
    async () => await allCharities(), []
  );

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        <h1>Error</h1>
        <div id="error-text">{error.toString()}</div>
      </div>
    );
  }

  return (
    <div>
      <div>
        <ul>
          {data.map((charity, index) => (
            <CharityCard key={index} charity={charity} />
          ))}
        </ul>
      </div>
    </div>
  );
}