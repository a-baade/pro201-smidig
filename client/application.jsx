import { BrowserRouter, Link, Route, Routes } from "react-router-dom";
import { FrontPage } from "./pages/frontPage";
import React, { useContext } from "react";
import { LoginPage } from "./pages/loginPage";
import { useLoading } from "./useLoading";
import { ApiContext } from "./apiContext";
import { Profile } from "./pages/profile";
import "./css/style.css";
import { Charities } from "./pages/charities";
import Navigation from "./components/navigation";
import Footer from "./components/footer";
import { RegisterNewUser } from "./pages/registerNewUser";
import { AboutUs } from "./pages/aboutUs";
import { SignUp } from "./pages/signUp";




export function Application() {
  const { fetchLogin } = useContext(ApiContext);
  const { data, error, loading, reload } = useLoading(fetchLogin);

  if (error) {
    return <div>Error: {error.toString()}</div>;
  }
  if (loading) {
    return <div>Please wait...</div>;
  }

  return (
    <BrowserRouter>
      <header>
        <Navigation/>
      </header>
      <main>
        <Routes>
          <Route path={"/"} element={<FrontPage />} />
          <Route path={"/charities"} element={<Charities />}/>
          <Route path={"/register"} element={<RegisterNewUser />}/>
          <Route path={"/about"} element={<AboutUs/>}/>
          <Route path={"/signup"} element={<SignUp config={data.config}/>}/>
          <Route
            path={"/login/*"}
            element={<LoginPage config={data.config} reload={reload} />}
          />
          <Route
            path={"/signup/*"}
            element={<LoginPage config={data.config} reload={reload} />}
          />
          <Route path={"/profile"} element={<Profile user={data?.user} />} />
          <Route path={"*"} element={<h1>Not found</h1>} />
        </Routes>
      </main>
      <Footer/>
    </BrowserRouter>
  );
}
