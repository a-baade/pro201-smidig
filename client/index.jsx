import React from "react";
import ReactDOM from "react-dom";
import {Application} from "./application";

ReactDOM.render(<Application />, document.getElementById("app"));