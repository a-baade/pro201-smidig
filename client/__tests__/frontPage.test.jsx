import { FrontPage } from "../frontPage";
import React from "react";
import { Link, MemoryRouter } from "react-router-dom";
import { createRoot } from "react-dom/client";

describe("FrontPage component", () => {
  it("should test  div element", async function () {
    const domEl = document.createElement("div");
    const root = createRoot(domEl);
    root.render(
      <MemoryRouter>
        <FrontPage />
      </MemoryRouter>
    );
    expect(domEl.innerHTML).toMatchSnapshot();
  });

  it("should render correctly", function() {
    const linkEl = document.createElement("Link");
    const root = createRoot(linkEl);
    root.render(
      <MemoryRouter>
        <Link to={"/login"}>Login</Link>
      </MemoryRouter>
    );
    expect(linkEl.innerHTML).toMatchSnapshot();
  });

  it("should test text in link element", function () {
    const linkEl = document.createTextNode(/log in/i);
    expect(linkEl).toMatchSnapshot();
  });

  it("should test link element", async function () {
    const linkEl = document.createElement("Link");
    const root = createRoot(linkEl);
    root.render(
      <MemoryRouter>
        <FrontPage />
      </MemoryRouter>
    );
    expect(linkEl.innerHTML).toMatchSnapshot();
  });
});
