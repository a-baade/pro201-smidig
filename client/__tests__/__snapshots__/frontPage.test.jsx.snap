// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontPage component should render correctly 1`] = `"<a href=\\"/login\\">Login</a>"`;

exports[`FrontPage component should test  div element 1`] = `"<div><h1>Front page</h1></div>"`;

exports[`FrontPage component should test link element 1`] = `"<div><h1>Front page</h1></div>"`;

exports[`FrontPage component should test text in link element 1`] = `/log in/i`;
